### ISSUE TITLE
Cursor color not updating with change in color before typing

### ISSUE THREAD

**Issue by ashu12chi** [ueberdosis/tiptap#3873]
### What’s the bug you are facing?

When I change color in my editor, the cursor's color is reflected according to the currently active color, which we also want. But this is not working when the color is changed and we have not typed anything.
In the below, GIF, we change color to purple, the cursor remains black until we typed anything, similarly, when we changed color to red, the cursor remains purple, until we changed color to red.
![Animation](https://user-images.githubusercontent.com/45908393/227150470-c88ac489-4256-400b-b568-eb2932d40cd7.gif)

### Which browser was this experienced in? Are any special extensions installed?

Chrome, Edge, Firefox. (Not tested on other browsers)

### How can we reproduce the bug on our side?

The bug is reproducible in example here: https://tiptap.dev/api/extensions/color
Follow same steps as shown in the GIF.

### Can you provide a CodeSandbox?

NA

### What did you expect to happen?

The cursor's color should change immediately.

### Anything to add? (optional)

NA

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Are you sponsoring us?

- [X] Yes, I’m a sponsor. 💖

**Comment by ashu12chi** [comment]
Commenting as the issue is still present.

**Comment by C-Hess** [comment]
I don't think this issue is fixable. When you are setting a color before typing anything, you're adding the text-color "mark" into what's called the `storedMarks` in editor state. This is how ProseMirror tracks what marks are "active" even though the marks don't exist in the document yet (until you start typing).

Because the document is not modified when `storedMarks` are used (because the marks aren't actually applied yet), the cursor color will not change.

**Comment by mnik01** [comment]
Hello!
As far as I can see, this can be quickly fixed with 
```css
caret-color: #000;
```

I've tested it on reproduction link placed into body inside of iframe with example:
```css
* {
caret-color: #000 !important;
}
```

And it worked

**Comment by brucecoad** [comment]
> Hello! As far as I can see, this can be quickly fixed with
> 
> ```css
> caret-color: #000;
> ```
> 
> I've tested it on reproduction link placed into body inside of iframe with example:
> 
> ```css
> * {
> caret-color: #000 !important;
> }
> ```
> 
> And it worked. See: ![image](https://private-user-images.githubusercontent.com/39025497/297374642-ddf3ab90-077d-404e-8067-30be7ff4ed0e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA0ODQwNDUsIm5iZiI6MTcxMDQ4Mzc0NSwicGF0aCI6Ii8zOTAyNTQ5Ny8yOTczNzQ2NDItZGRmM2FiOTAtMDc3ZC00MDRlLTgwNjctMzBiZTdmZjRlZDBlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE1VDA2MjIyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUwNDU5ZDQ4MWQ1ZjdmNTdjN2U3NGFlN2E1NTk2OWQ1Mzk5NzBiZGM4OTkzNDdiMDk3ZTI5OTFkZDEwYzg5MmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NcfpRoyLV9B7TiNsl-3VDMExJNuVuqfadbjxK4-JFKY)

Thank you. I was having the same problem, but that fixed it.

**Comment by bdbch** [comment]
Closing this for now - if this is a massive bug or serious issue, feel free to ping us and we can look into this again. Since we have a workaround this should and has a very low impact area for users I wouldn't keep this open for a fix for now.
